main.md

```diff
@@ -401,6 +435,56 @@ known type but containing unknown elements or elements of the wrong type, the AS
 
 # Token Request
 
+The `authorization_details` request parameter can be used to specify the authorization details a client wants the AS to assign to an access token. The AS checks whether the underlying grant (in case of grant types `authorization_code`, `refresh_token`, ...) or the client's policy (in case of grant type `client_credential`) allows the issuance of an access token with the requested authorization details. Otherwise, the AS refuses the request with error code `invalid_authorization_details` (similar to `invalid_scope`).
+
+## Comparing authorization details
+
+Many actions in the OAuth protocol allow the AS and RS to make security decisions based on whether or not the request
+is asking for "more" or "less" than a previous, existing request. For example, upon refreshing a token, the client can
+ask for a new access token with "fewer permissions" than had been previously authorized by the resource owner.
+Since the nature of an authorization details request is based solely on the API or APIs that it is describing, there is not
+a simple means of comparing any two arbitrary authorization details requests. 
+Authorization servers should not rely on simple object comparison in most cases, as the intersection of some elements
+within a request could have side effects in the access rights granted, depending on how the API
+has been designed and deployed. This is a similar effect to the scope values used with some APIs.
+
+However, when comparing a new request to an existing request, authorization servers can use the same 
+processing techniques as used in granting the request in the first place to determine if a resource
+owner needs to authorize the request. 
+
+This shall be illustrated using our running example. The example authorization request in (#authz_request), if approved by the user, resulted in the issuance of an authorization code associated with the privileges to 
+
+* list accounts 
+* access the balance of one or more accounts,
+* access the transactions of one or more accounts, and 
+* to initiate a payment. 
+
+The client could now request the AS to issue an access token assigned with the privilege to just access a list of account as follows:
+
+```JSON
+[
+   {
+      "type":"account_information",
+      "actions":[
+         "list_accounts"
+      ],
+      "locations":[
+         "https://example.com/accounts"
+      ]
+   }
+]
+```
+
+The AS would 
+
+* compare that the authorization code issued in the previous step contains an authorization details object of type `account_information`
+* compare whether the approved list of actions contains `list_account`, and 
+* whether the `locations` value matches.  
+
+If all checks succeed, the AS would issue the requested access token. 
```

Comment:
This paragraph feels more prescriptive than we mean for it to be, I think. While it's in an example section, as the only example this list of actions feels like it takes on more weight than intended. If we're going to give an example like this, we should show another example that does a different style of processing. The danger here is that developers start to make dangerous assumptions about what is essentially an internal process, and base their security decisions on that.

Reply:
I could replace this text by a more general explanation that a comparison will either check for equivalence or a subset relationship on the level of individual fields and give some examples, locations (subset), identifier (match), ...
WDYT?

Reply:
@tlodderstedt updated text, please review